src/commands/ext-configure.ts

```diff
@@ -76,9 +76,10 @@ export const command = new Command("ext:configure <extensionInstanceId>")
       instanceId,
       projectDir: config.projectDir,
     });
+    const params = (spec.params ?? []).concat(spec.systemParams ?? []);
     const [immutableParams, tbdParams] = partition(
-      (spec.params ?? []).concat(spec.systemParams ?? []),
-      (param) => param.immutable ?? false
+      params,
+      (param) => (param.immutable && !!oldParamValues[param.param]) ?? false
```

Comment:
Is this the same as `!!(param.immutable && oldParamValues[param.param])`

Reply:
I found it to be a little easier to reason about than the `boolean|undefine && boolean` (TIL this can be undefined..)

Reply:
It is not - TS will automatically cast (string) to (bool) in an if statement, but the && operator will not